src/distro/distro.py

```diff
@@ -1078,6 +1084,53 @@ def uname_attr(self, attribute: str) -> str:
         """
         return self._uname_info.get(attribute, "")
 
+    def __resolve_chroot_symlink(self, link_location: str) -> str:
+        # only resolve symlink if root_dir is set and link is under it
+        if (
+            self.root_dir is None
+            or os.path.commonprefix([self.root_dir, link_location]) != self.root_dir
+        ):
+            return link_location
+
+        seen_paths = []
+        while True:
+            try:
+                resolved = os.readlink(link_location)
+            except OSError:  # includes case "not a symlink"
+                if (
+                    os.path.commonprefix(
+                        [
+                            os.path.realpath(self.root_dir),
+                            os.path.realpath(link_location),
+                        ]
+                    )
+                    != os.path.realpath(self.root_dir)
+                ):
+                    raise PermissionError(
```

Comment:
Maybe not the most important thing in the world, but `FileNotFoundError` seems to be a better match here because:

```console
# cd "$(mktemp -d)"
# ln -s no-such-file broken-symlink
# python -c 'open("broken-symlink")' |& tail -n1
FileNotFoundError: [Errno 2] No such file or directory: 'broken-symlink'
```

From the point of view of inside the chroot it's not a permission error. Are you sure you want to go with `PermissionError`?